lib/webviewfetchinterceptor/src/main/java/eu/kanade/tachiyomi/lib/webviewfetchinterceptor/WebViewFetchInterceptor.kt

```diff
@@ -0,0 +1,358 @@
+                if (userAgent && userAgent.length > 0) {
+                    headers.set('User-Agent', userAgent);
+                }
+
+                // Perform fetch
+                fetch(requestUrl, {
+                    method: requestMethod,
+                    headers: headers,
+                    body: body,
+                    credentials: 'include',
+                    mode: 'cors'
+                })
+                .then(async (response) => {
+                    // Read body as ArrayBuffer
+                    const arrayBuffer = await response.arrayBuffer();
+
+                    // Convert ArrayBuffer to base64
+                    const bytes = new Uint8Array(arrayBuffer);
+                    let binary = '';
+                    for (let i = 0; i < bytes.length; i++) {
+                        binary += String.fromCharCode(bytes[i]);
+                    }
+                    const bodyBase64 = btoa(binary);
+
+                    // Convert headers to JSON string
+                    const headersObj = {};
+                    response.headers.forEach((value, key) => {
+                        headersObj[key] = value;
+                    });
+                    const headersJson = JSON.stringify(headersObj);
+
+                    // Call Android interface
+                    window.android.onResponse(
+                        response.status,
+                        response.statusText,
+                        headersJson,
+                        bodyBase64
+                    );
+                })
+                .catch((error) => {
+                    window.android.onError(error.toString());
+                });
+            })();
+        """.trimIndent()
+
+        handler.post {
+            val webview = WebView(context)
+            webView = webview
+
+            with(webview.settings) {
+                javaScriptEnabled = true
+                domStorageEnabled = true
+                databaseEnabled = true
+                useWideViewPort = false
+                loadWithOverviewMode = false
+                userAgentString = request.header("User-Agent")
+            }
+
+            webview.addJavascriptInterface(jsInterface, "android")
+
+            webview.webViewClient = object : WebViewClient() {
+                override fun onPageFinished(view: WebView, url: String) {
+                    // Execute script after page loads
+                    view.evaluateJavascript(jsScript) { result ->
+                        // Handle JavaScript errors here
+                        if (result == null) {
+                            Log.e("WebViewFetchInterceptor", "JavaScript evaluation returned null")
+                            jsInterface.onError("Error executing JavaScript script")
+                        }
+                    }
+                }
+            }
+
+            // Load baseUrl first to establish context
+            webview.loadUrl(loadUrl)
```

Comment:
```suggestion
            // establish context
            webview.loadDataWithBaseURL("$baseUrl/", " ", "text/html", null, null)
```
this will not load any page but still establish correct url context

Reply:
A big thanks about it, I update with a solution to extract the base url from current request